pkg/pkgmgr/rpm.go

```diff
@@ -449,14 +457,18 @@ func (rm *rpmManager) installUpdates(ctx context.Context, updates unversioned.Up
 	// Install patches using available rpm managers in order of preference
 	var installCmd string
 	switch {
-	case rm.rpmTools["dnf"] != "":
-		checkUpdateTemplate := `sh -c "%[1]s check-update; if [ $? -ne 0 ]; then echo >> /updates.txt; fi"`
-		if !rm.checkForUpgrades(ctx, rm.rpmTools["dnf"], checkUpdateTemplate) {
+	case rm.rpmTools["tdnf"] != "" || rm.rpmTools["dnf"] != "":
+		dnfTooling := rm.rpmTools["tdnf"]
+		if dnfTooling == "" {
```

Comment:
what is this check case for?

Reply:
existing code didn't account for tdnf. AL3 doesn't have dnf or yum, so it was failing to find the proper tooling. in this case, we are checking if tdnf is present. if not, falling back to dnf.